ledger/src/blockstore.rs

```diff
@@ -1493,22 +1525,94 @@ impl Blockstore {
         shred_index < slot_meta.consumed || data_index.contains(shred_index)
     }
 
-    fn get_data_shred_from_just_inserted_or_db<'a>(
+    fn get_shred_from_just_inserted_or_db<'a>(
         &'a self,
         just_inserted_shreds: &'a HashMap<ShredId, Shred>,
         slot: Slot,
-        index: u64,
+        index: u32,
+        shred_type: ShredType,
     ) -> Cow<'a, Vec<u8>> {
-        let key = ShredId::new(slot, u32::try_from(index).unwrap(), ShredType::Data);
+        let key = ShredId::new(slot, index, shred_type);
         if let Some(shred) = just_inserted_shreds.get(&key) {
             Cow::Borrowed(shred.payload())
-        } else {
+        } else if shred_type == ShredType::Data {
```

Comment:
Can we use a `match` expression here?
This code is not safe if `ShredType` enum is extended with a new variant. (unlikely that it happens but no reason to leave this foot gun here).